src/Compilers/CSharp/Portable/Symbols/Attributes/AttributeData.cs

```diff
@@ -256,25 +256,37 @@ static internal void DecodeSkipLocalsInitAttribute<T>(CSharpCompilation compilat
         static internal void DecodeMemberNotNullAttribute<T>(TypeSymbol type, ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
             where T : WellKnownAttributeData, IMemberNotNullAttributeTarget, new()
         {
-            var membersArray = arguments.Attribute.CommonConstructorArguments[0];
-            if (membersArray.IsNull)
+            var value = arguments.Attribute.CommonConstructorArguments[0];
+            if (value.IsNull)
             {
                 return;
             }
 
-            var builder = ArrayBuilder<string>.GetInstance();
-            foreach (var member in membersArray.Values)
+            if (value.Kind != TypedConstantKind.Array)
             {
-                var memberName = member.DecodeValue<string>(SpecialType.System_String);
+                string? memberName = value.DecodeValue<string>(SpecialType.System_String);
```

Comment:
Won't this throw in the case where the customer uses a version of `MemberNotNull` where there is a ctor with a non-string argument? I'm unsure what our tolerance here is for malformed attributes. Know @RikkiGibson is hitting this with the obsolete change.

Reply:
We should get here in such case. This method is only called once we've found the attribute we're interested in by AttributeDescription (which verifies one of two signatures we want was matched)